File: dotnet/tests/Obscura.Browser.Tests/InputAccept.cs
using Xunit;

namespace Obscura.Browser.Tests;

/// <summary>
/// The xUnit port of <c>crates/obscura-browser/tests/input_accept.rs</c>.
/// </summary>
public sealed class InputAcceptTests
{
    [Fact]
    public async Task InputAcceptReflectsTheContentAttribute()
    {
        BrowserContext context = BrowserContext.WithStorageAndNetwork(
            "input-accept", null, false, null, null, true);
        using var page = new Page("input-accept-page", context);
        await page.NavigateAsync(
            "data:text/html,<input id=upload type=file accept='image/png,image/jpeg'>");

        PageFixtures.AssertJson(
            """
            {
                "initial": "image/png,image/jpeg",
                "property": "image/webp",
                "attribute": "image/webp"
            }
            """,
            page.Evaluate(
                """
                (() => {
                    const input = document.getElementById('upload');
                    const initial = input.accept;
                    input.accept = 'image/webp';
                    return {
                        initial,
                        property: input.accept,
                        attribute: input.getAttribute('accept'),
                    };
                })()
                """));
    }
}

File: .reference/obscura/crates/obscura-browser/tests/input_accept.rs
use std::sync::Arc;

use obscura_browser::{BrowserContext, Page};

#[tokio::test(flavor = "current_thread")]
async fn input_accept_reflects_the_content_attribute() {
    let context = Arc::new(BrowserContext::with_storage_and_network(
        "input-accept".to_owned(),
        None,
        false,
        None,
        None,
        true,
    ));
    let mut page = Page::new("input-accept-page".to_owned(), context);
    page.navigate("data:text/html,<input id=upload type=file accept='image/png,image/jpeg'>")
        .await
        .unwrap();

    assert_eq!(
        page.evaluate(
            r#"
            (() => {
                const input = document.getElementById('upload');
                const initial = input.accept;
                input.accept = 'image/webp';
                return {
                    initial,
                    property: input.accept,
                    attribute: input.getAttribute('accept'),
                };
            })()
            "#,
        ),
        serde_json::json!({
            "initial": "image/png,image/jpeg",
            "property": "image/webp",
            "attribute": "image/webp",
        })
    );
}
